### PR TITLE
mtl: add Aria module to build

### PR DESCRIPTION
**Pull request by pblaszko**
Add Aria module to MTL build.

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]
Simple PR, but I guess you are still testing hence draft.

**Comment by pblaszko** [comment]
> Simple PR, but I guess you are still testing hence draft.

Trying to figure-out problems with different compilers than I have locally.

**Comment by marc-hb** [comment]
> Simple PR, but I guess you are still testing hence draft.

Github quirk: PRs have to be CREATED as draft to avoid spamming reviewers.

I always create all my PRs as draft. There is zero drawback, only advantages.

Unsubscribing now :-)

**Review by abonislawski — APPROVED** [review]

**Comment by pblaszko** [comment]
> > Simple PR, but I guess you are still testing hence draft.
> 
> Github quirk: PRs have to be CREATED as draft to avoid spamming reviewers.
> 
> I always create all my PRs as draft. There is zero drawback, only advantages.
> 
> Unsubscribing now :-)

[marc-hb](https://github.com/marc-hb) lessons learned for next PR's :)

**Comment by abonislawski** [comment]
@wszypelt please add Aria test to CI when merged
if it still works ofc ;-)

**Comment by pblaszko** [comment]
The problem with building other configuration was related to missing non-HIFI implementation.
Adding generic processing functions.

File: src/audio/aria/aria_generic.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	}
+	smpl_groups = size / cd->chan_cnt;
+	step = (gain_end - gain_begin) / smpl_groups;
+
+	for (size_t idx = 0; idx < smpl_groups; ++idx) {
+		gain = gain_begin + idx * step;
+		ii = (cd->buff_pos + cd->chan_cnt * idx) %  cd->buff_size;
+		for (size_t ch = 0; ch < cd->chan_cnt; ++ch) {
+			sample = cd->data[ii + ch];
+			data[cd->chan_cnt * idx + ch] = (sample * gain) >> 32;
```
Comment (lyakh) [inline src/audio/aria/aria_generic.c:51]: you could speed it up a bit by eliminating 2 multiplications by `idx`

Comment (pblaszko) [reply src/audio/aria/aria_generic.c]: Thanks! Done.

**Review by dnikodem — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Comment by lgirdwood** [comment]
@wszypelt good for you or are we failing the new test ?

File: src/audio/aria/aria_generic.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		if (cd->gains[(cd->gain_state + i + 3) % ARIA_MAX_GAIN_STATES] < gain_end)
+			gain_end = cd->gains[(cd->gain_state + i + 3) % ARIA_MAX_GAIN_STATES];
+	}
+	smpl_groups = size / cd->chan_cnt;
+	step = (gain_end - gain_begin) / smpl_groups;
+
+	for (size_t idx = 0; idx < smpl_groups; ++idx) {
+		gain = gain_begin + idx * step;
+		ii = (cd->buff_pos + cd->chan_cnt * idx) %  cd->buff_size;
+		offset = cd->chan_cnt * idx;
```
Comment (lyakh) [inline src/audio/aria/aria_generic.c:48]: if you calculate `cd->chan_cnt * idx` in a temporary variable, you could as well use it in the line above

File: src/audio/aria/aria_generic.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		for (size_t ch = 0; ch < cd->chan_cnt; ++ch) {
+			sample = cd->data[ii + ch];
+			data[offset + ch] = (sample * gain) >> 32;
+		}
+	}
```
Comment (lyakh) [inline src/audio/aria/aria_generic.c:53]: I meant replacing multiplications with additions:
```
	for (size_t idx = 0, gain = gain_begin, offset = 0;
	     idx < smpl_groups;
	     ++idx, gain += step, offset += cd->chan_cnt) {
		ii = (cd->buff_pos + offset) % cd->buff_size;
		for (size_t ch = 0; ch < cd->chan_cnt; ++ch) {
			sample = cd->data[ii + ch];
			data[offset + ch] = (sample * gain) >> 32;
		}
	}

**Review by lyakh — CHANGES_REQUESTED** [review]

**Comment by pblaszko** [comment]
> @wszypelt good for you or are we failing the new test ?

Very weird: consequently some simple "DMIC->Copier->Copier->Host" test fails, while at the same time on other PR's it is passing. Logically it is completely not related to changes from this PR... I need to check it locally.

**Comment by wszypelt** [comment]
@RanderWang 
I'm making changes now with log copying and some PRs may have been affected, within 10 minutes you should have the latest results, you can fully view them on the internal site. I am in the process of preparing an e-mail with a description

**Review by lyakh — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Pull request merged**